util/deephash/deephash.go

```diff
@@ -136,7 +137,21 @@ func print(w *bufio.Writer, v reflect.Value, visited map[uintptr]bool, scratch [
 		return acyclic
 	case reflect.Slice, reflect.Array:
 		if v.Type().Elem().Kind() == reflect.Uint8 && v.CanInterface() {
-			fmt.Fprintf(w, "%q", v.Interface())
+			if v.CanAddr() {
+				Len := v.Len()
+				w.Write(strconv.AppendInt(scratch[:0], int64(Len), 10))
+				w.WriteByte(':')
+				if Len > 0 {
+					var s []byte
+					sh := (*reflect.SliceHeader)(unsafe.Pointer(&s))
+					sh.Data = v.Addr().Pointer()
+					sh.Len = Len
+					sh.Cap = Len
+					w.Write(s) // unsafe slice escapes, but w is trusted *bufio.Writer
+				}
+			} else {
+				fmt.Fprintf(w, "%q", v.Interface())
```

Comment:
So the old way of printing with `%q` provides some protection against intentional collisions because `"` acts as a delimiter that is escaped if it appears inside the byte array. The preceding length may still provide that same protection but I'm less convinced about it. How important is this to worry about?

Reply:
It's a valid concern. I thought I'd addressed it, but I just made it more explicit in the latest round. All variable length items have a length first now.